pkg/backend/remote/remote.go

```diff
@@ -302,6 +305,24 @@ func (r *Remote) monitorPing(client *dataconn.Client) {
 	}
 }
 
+func (r *Remote) pingWithTimeout(client *dataconn.Client) error {
+	timeout := time.After(pingTimeout)
+	err := ErrPingTimeout
+
+	status := make(chan error, 1)
```

Comment:
Allocating a channel every 2 seconds? It is hard work for GC.

Reply:
The time.After was making a channel too.  The only thing I could think of it reusing the ticker for pings for determining a ping timeout which doesn't create new channels every iteration.